chef/cookbooks/provisioner/recipes/update_nodes.rb

```diff
@@ -220,21 +220,32 @@
         end
 
         case
-        when os =~ /^ubuntu/
-          append << "url=#{node_url}/net_seed"
-          template "#{node_cfg_dir}/net_seed" do
+        when os =~ /^debian/ || os =~ /^ubuntu/
+          append << "auto=true url=#{node_url}/preseed.cfg"
+          # The values in the preseed file can't work as they're fetched after network is up
+          append << "hostname=#{mnode[:hostname]} domain=#{mnode[:domain]}"
+
+          ntp_server = search(:node, "roles:ntp-server").first
+          ntp_server_ip = Chef::Recipe::Barclamp::Inventory.get_network_by_type(ntp_server, "admin").address unless ntp_server.nil?
```

Comment:
Metrics/LineLength: Line is too long. [131/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)